app/src/features/projects/view/ProjectPage.test.tsx

```diff
@@ -111,6 +151,186 @@ describe('ProjectPage', () => {
     });
   });
 
+  it('sees delete project button as disabled when accessing a published project as a project administrator', async () => {
+    mockBiohubApi().codes.getAllCodeSets.mockResolvedValue({
+      activity: [{ id: 1, name: 'activity 1' }]
+    } as any);
+    mockBiohubApi().project.getProjectForView.mockResolvedValue(getProjectForViewResponse);
+    mockBiohubApi().project.deleteProject.mockResolvedValue(true);
+
+    const hasSystemRole = (validSystemRoles?: string[]) => {
+      if (!validSystemRoles || !validSystemRoles.length) {
+        return true;
+      }
+
+      const userSystemRoles = authState.keycloakWrapper.systemRoles;
+
+      for (const validRole of validSystemRoles) {
+        if (userSystemRoles.includes(validRole)) {
+          return true;
+        }
+      }
+
+      return false;
+    };
+
+    const authState = {
+      keycloakWrapper: {
+        keycloak: {
+          authenticated: true
+        },
+        hasLoadedAllUserInfo: true,
+        systemRoles: [SYSTEM_ROLE.PROJECT_ADMIN] as string[],
+        getUserIdentifier: () => 'testuser',
+        hasAccessRequest: false,
+        hasSystemRole: hasSystemRole,
+        getIdentitySource: () => 'idir',
+        username: 'testusername',
+        displayName: 'testdisplayname',
+        email: 'test@email.com',
+        firstName: 'testfirst',
+        lastName: 'testlast',
+        refresh: () => {}
+      }
+    };
+
+    const { getByTestId, findByText, getByText } = render(
+      <AuthStateContext.Provider value={authState}>
+        <DialogContextProvider>
+          <Router history={history}>
+            <ProjectPage />
+          </Router>
+        </DialogContextProvider>
+      </AuthStateContext.Provider>
+    );
+
+    const projectHeaderText = await findByText('Test Project Name', { selector: 'h1' });
+    expect(projectHeaderText).toBeVisible();
+
+    expect(getByTestId('delete-project-button')).toBeDisabled();
+  });
+
+  it('sees delete project button as enabled when accessing an unpublished project as a project administrator', async () => {
+    mockBiohubApi().codes.getAllCodeSets.mockResolvedValue({
+      activity: [{ id: 1, name: 'activity 1' }]
+    } as any);
+    mockBiohubApi().project.getProjectForView.mockResolvedValue({
+      ...getProjectForViewResponse,
+      project: { ...getProjectForViewResponse.project, publish_date: '' }
+    });
+    mockBiohubApi().project.deleteProject.mockResolvedValue(true);
+
+    const hasSystemRole = (validSystemRoles?: string[]) => {
+      if (!validSystemRoles || !validSystemRoles.length) {
+        return true;
+      }
+
+      const userSystemRoles = authState.keycloakWrapper.systemRoles;
+
+      for (const validRole of validSystemRoles) {
+        if (userSystemRoles.includes(validRole)) {
+          return true;
+        }
+      }
+
+      return false;
+    };
```

Comment:
why is this function in the tests?
also can it be placed at top of file so it can be reused by many tests?